import/WebView.qml

```diff
@@ -0,0 +1,286 @@
+                break
+            }
+            default: {
+                console.log("unhandled async message received: " + message)
+                break
+            }
+        }
+    }
+
+    BusyIndicator {
+        id: busySpinner
+        anchors.centerIn: parent
+        running: true
+        visible: webview.loading
+        size: BusyIndicatorSize.Large
+    }
+
+    Timer {
+        id: helper
+        property var geolocationUrls: {{}}
+        property var contextMenu
+
+        function uniqueId() {
+            return Math.floor((Math.random() * 10000000) + 1)
+        }
+
+        function hideVirtualKeyboard() {
+            if (Qt.inputMethod.visible) {
+                webview.focus = true
+            }
+        }
+
+        function openAuthDialog(webView, data, winid) {
+            if (pageStack.busy) {
+                helper.delayedOpenAuthDialog(webView, data, winid)
+            } else {
+                helper.immediateOpenAuthDialog(webView, data, winid)
+            }
+        }
+
+        function immediateOpenAuthDialog(webView, data, winid) {
+            var dialog = pageStack.push(authDialog,
+                                        {"hostname": data.text, "realm": data.title,
+                                         "username": data.storedUsername, "password": data.storedPassword,
+                                         "passwordOnly": data.passwordOnly })
+            dialog.accepted.connect(function () {
+                webView.sendAsyncMessage("authresponse",
+                                         { "winid": winid, "accepted": true,
+                                           "username": dialog.username, "password": dialog.password,
+                                           "dontsave": dialog.dontsave })
+            })
+            dialog.rejected.connect(function() {
+                webView.sendAsyncMessage("authresponse",
+                                         { "winid": winid, "accepted": false})
+            })
+        }
+
+        property var authDialogWebView
+        property var authDialogData
+        property var authDialogWinId
+        function delayedOpenAuthDialog(webView, data, winid) {
+            authDialogWebView = webView
+            authDialogData = data
+            authDialogWinId = winid
+            start()
+        }
+
+        repeat: false
+        running: false
+        interval: 600 // page transition delay.
+        onTriggered: {
+            openAuthDialog(authDialogWebView, authDialogData, authDialogWinId)
+        }
+
+        function findParentWithProperty(item, propertyName) {
+            var parentItem = item.parent
+            while (parentItem) {
+                if (parentItem.hasOwnProperty(propertyName)) {
+                    return parentItem
+                }
+                parentItem = parentItem.parent
+            }
+            return null
+        }
+
+        function setActiveInPage() {
+            if (webview.active) {
```

Comment:
There could be a ```webViewPage``` property that could be also explicitly set. Futhermore, there is no need to lookup again if ```webViewPage``` has been already found. I don't think that it is a good idea to allowing moving webview between Silica Pages. So, lookup only once.

Background: when screen is blanked we should suspend the view conditionally. As a result active changes as well. So when display is turned on again view should become active again which should not result webViewPage lookup.